modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -816,6 +816,23 @@ public void postProcessParameter(CodegenParameter parameter) { }
               itr.remove();
             }
         }
+
+        @SuppressWarnings("unchecked")
+        Map<String, Object> operations = (Map<String, Object>) objs.get("operations");
+        if (operations != null) {
+
+            @SuppressWarnings("unchecked")
+            List<CodegenOperation> ops = (List<CodegenOperation>) operations.get("operation");
+            for (CodegenOperation operation : ops) {
+
+                if (operation.returnType == null) {
+                    operation.returnType = "void";
```

Comment:
The code for `sync` call will also break:
```
l, null);
-        return apiClient.execute(call);
+        Type localVarReturnType = new TypeToken<void>(){}.getType();
+        return apiClient.execute(call, localVarReturnType);
     }
```

@jfiala what about doing it in the Java-related generators one by one instead of doing it in "abstract" Java generator.

For Java API client, any concern to keep the current approach? Can you also remind me the limitation with the current approach?